pallets/api/src/messaging/mod.rs

```diff
@@ -399,8 +437,8 @@ pub mod pallet {
 			T::XcmResponseOrigin::ensure_origin(origin)?;
 
 			// Lookup message from query id.
-			let (origin, id) = XcmQueries::<T>::take(query_id).ok_or(Error::<T>::InvalidQuery)?;
-			let Some(Message::XcmQuery { query_id, callback, deposit }) =
+			let (origin, id) = XcmQueries::<T>::get(query_id).ok_or(Error::<T>::InvalidQuery)?;
```

Comment:
any reason you are wanting to leave the xcm query in the storage? (no longer using `take`).